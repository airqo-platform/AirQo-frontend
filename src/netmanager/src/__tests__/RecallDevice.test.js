import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import DeviceDeployStatus from '../views/components/DataDisplay/DeviceView/DeviceDeployStatus';

jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn(),
  }));

  // i have used the DeviceDeployStatus for the tests because all the functions for the recall an deploy are in 
  // this component
  
  describe('RecallDevice', () => {
    it('renders without errors', () => {
      const { asFragment } = render(
        <DeviceDeployStatus deviceData={{}} siteOptions={[]} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
