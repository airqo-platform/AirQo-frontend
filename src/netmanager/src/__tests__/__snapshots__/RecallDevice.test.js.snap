// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecallDevice renders without errors 1`] = `
<DocumentFragment>
  <div
    class="loader-container"
    style="z-index: -1; position: fixed; top: 0px; left: 0px; right: 0px; height: 4px;"
  />
  <div
    style="display: flex; flex-wrap: wrap; padding: 40px 0px 10px 0px; margin: 0px auto; align-items: baseline; justify-content: flex-end;"
  >
    <span
      style="display: flex; align-items: bottom; justify-content: flex-end;"
    >
      <span
        style="display: flex; align-items: center; font-size: 1.2rem; margin-right: 10px;"
      >
        <span
          style="font-size: 0.7rem; margin-right: 10px; background: rgb(255, 255, 255); border: 1px solid #ffffff; border-radius: 5px; padding: 0px 5px;"
        >
          Deploy status
        </span>
         
        <span
          style="color: red; text-transform: capitalize;"
        >
          not deployed
        </span>
      </span>
      <div
        style="max-width: 500px; width: 100%;"
      >
        <div
          class="dropdown-rapper-recall"
        />
        <span
          class=""
          title="Recall Device"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Recall Device
            </span>
          </button>
        </span>
      </div>
    </span>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
    style="margin: 0px auto; min-height: 400px; padding: 20px 20px; max-width: 1500px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          style="margin-bottom: 15px;"
        >
          <div
            style="width: 100%;"
          >
            <div
              class=" css-2b097c-container"
            >
              <div
                class=" css-yk16xz-control"
              >
                <div
                  class=" css-1ma7x3-ValueContainer"
                >
                  <div
                    class=" css-25rwlf-placeholder"
                  >
                    <span
                      style="position: relative; background-color: white; opacity: 1; padding: 0px 5px;"
                    >
                      Site
                    </span>
                  </div>
                  <div
                    class=" css-1uccc91-singleValue"
                  />
                  <div
                    class="css-b8ldur-Input"
                  >
                    <div
                      class=""
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-2-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          style="margin-bottom: 15px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
          >
            Height
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            style="width: 100%; height: 100%;"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              required=""
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-233 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-235"
              >
                <span>
                  Height *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          style="margin-bottom: 15px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
          >
            Power type
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            style="width: 100%; height: 50px;"
          >
            <select
              aria-invalid="false"
              class="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              required=""
            >
              <option
                value=""
              />
              <option
                value="Mains"
              >
                Mains
              </option>
              <option
                value="Solar"
              >
                Solar
              </option>
              <option
                value="Alternator"
              >
                Alternator
              </option>
            </select>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-233 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-235"
              >
                <span>
                  Power type *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          style="margin-bottom: 15px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink="false"
          >
            Mount Type
            <span
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            style="width: 100%; height: 50px;"
          >
            <select
              aria-invalid="false"
              class="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              required=""
            >
              <option
                value=""
              />
              <option
                value="Faceboard"
              >
                Faceboard
              </option>
              <option
                value="Pole"
              >
                Pole
              </option>
              <option
                value="Rooftop"
              >
                Rooftop
              </option>
              <option
                value="Suspended"
              >
                Suspended
              </option>
              <option
                value="Wall"
              >
                Wall
              </option>
            </select>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-233 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-235"
              >
                <span>
                  Mount Type *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
          style="margin-bottom: 15px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-focused Mui-focused"
            data-shrink="true"
            for="deployment_date"
            id="deployment_date-label"
          >
            Deployment Date
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="deployment_date"
              type="date"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-233 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-235 PrivateNotchedOutline-legendNotched-236"
              >
                <span>
                  Deployment Date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          style="margin: 30px 0px 20px 0px;"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item MuiGrid-grid-xs-12"
          >
            <label
              class="MuiFormControlLabel-root"
              style="margin: 10px 0px 0px 5px; width: 100%;"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-268 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-271"
                    data-indeterminate="false"
                    name="primaryDevice"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Primary device in this site
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root"
              style="margin-left: 5px;"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-268 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-271"
                    data-indeterminate="false"
                    name="collocation"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Formal collocation in this site
              </span>
            </label>
          </div>
           
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end MuiGrid-align-content-xs-flex-end MuiGrid-justify-xs-flex-end MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <span
          class=""
          title="Run device test to activate"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            style="margin-left: 10px;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Deploy
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
